pkg/sql/parser/indexed_vars.go

```diff
@@ -238,3 +254,19 @@ func (h *IndexedVarHelper) VisitPre(expr Expr) (recurse bool, newExpr Expr) {
 
 // VisitPost implements the Visitor interface.
 func (*IndexedVarHelper) VisitPost(expr Expr) Expr { return expr }
+
+type unboundContainerType struct{}
+
+var unboundContainer = &unboundContainerType{}
```

Comment:
Please add a comment explaining when this is used, and how it differs from a `nil` container. I think the gist of it is that it indicates that we were previously bound to a container but we aren't anymore, right?